src/control/PlanarArmController.h

```diff
@@ -148,14 +149,20 @@ class PlanarArmController {
 	Eigen::Vector2d normalizeEEWithinRadius(Eigen::Vector2d eePos) {
 		double radius = kin.getSegLens().sum() * safetyFactor;
 
-		if (eePos.norm() > radius) {
-			// TODO: will need to eventually shrink velocity vector until it is within radius
-			// instead of just normalizing it.
-
-			eePos.normalize();
-			eePos *= radius;
+		if (eePos.norm() > (radius + 0.5)) {
+			double radius = kin.getSegLens().sum();
```

Comment:
This line declares `radius` to an incorrect value (we need it to be multiplied by the safety factor). Just remove this line and it should be fixed. Not you'll have to change the expected value in the test case.